glean-core/metrics.yaml

```diff
@@ -320,3 +320,25 @@ glean.error:
     expires: never
     no_lint:
       - COMMON_PREFIX
+
+glean.upload:
+  ping_upload_failure:
+    type: labeled_counter
+    description:
+      Counts the number of times pings fail to upload
+      and the reason why they failed.
```

Comment:
Maybe it's worth mentioning that (obviously) these numbers are reported after the fact?
That is: if a metrics ping upload fails and gets retried it still won't have this in there, only the next one will.

Reply:
Maybe it could be confusing mention this, because this metric will count the failures for all ping uploads, not only the metrics ping. It could be misleading to mention this situation
> if a metrics ping upload fails and gets retried it still won't have this in there, only the next one will.

because it only applies for the metrics ping.

Reply:
Hm, @mdboom, got an idea how to phrase this?
I'd like us to start being a bit more explicit of when and where to expect this metric

Reply:
It's not great -- this is the best I could come up with.  What do you think?

```
Counts the number of ping upload failures, by type of failure.
This includes failures for all ping types, though the counts appear in the next successfully sent `metrics` ping.
```